src/core/trtserver.h

```diff
@@ -376,6 +377,96 @@ TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_TraceNew(
 /// \return a TRTSERVER_Error indicating success or failure.
 TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_TraceDelete(TRTSERVER_Trace* trace);
 
+/// Get the activity userp of the trace object, which is supplied in the call to
+/// TRTSERVER_TraceNew.
+/// \param trace The trace object.
+/// \param userp Returns the activity userp of the trace object.
+/// \return a TRTSERVER_Error indicating success or failure.
+TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_TraceUserp(
```

Comment:
I don't think we need this function so I'd rather we don't have it.

Reply:
This is used because the user provided trace manager needs to get the tracer associated with the trace to free it in the release function. But as you said, I think this can be remove as we are going to pass the userp argument to the release function.